zookeeper-server/src/main/java/org/apache/zookeeper/server/backup/RestoreFromBackupTool.java

```diff
@@ -290,9 +299,14 @@ public boolean runWithRetries(CommandLine cl) {
         return true;
       } catch (IllegalArgumentException re) {
         System.err.println(
-            "Restore attempt failed due to insufficient backup files; attempting again. " + tries
-                + "/" + MAX_RETRIES + ". Error message: " + re.getMessage());
-        break;
+            "Restore attempt failed due to insufficient backup files. Error message: "
+                + re.getMessage());
+        return false;
+      } catch (BackupException be) {
+        System.err.println(
+            "Restoration attempt failed due to a backup exception, please check the environment and try again. Error message: "
```

Comment:
"check the environment" is a little unclear. Did you mean to say this was caused by an I/O error?

Reply:
No it's not caused by I/O error. It usually means some directories do not exist or failed to create such directories. I updated the message.